{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: g2papi in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (1.0.4)\n",
      "Requirement already satisfied: requests in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from g2papi) (2.28.1)\n",
      "Requirement already satisfied: pandas in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from g2papi) (1.5.2)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from pandas->g2papi) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from pandas->g2papi) (2022.6)\n",
      "Requirement already satisfied: numpy>=1.21.0 in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from pandas->g2papi) (1.23.5)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from requests->g2papi) (2.1.1)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from requests->g2papi) (3.4)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from requests->g2papi) (1.26.12)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from requests->g2papi) (2022.9.24)\n",
      "Requirement already satisfied: six>=1.5 in /Users/jsafer/.myenvs/jupyter_default/lib/python3.10/site-packages (from python-dateutil>=2.8.1->pandas->g2papi) (1.16.0)\n"
     ]
    }
   ],
   "source": [
    "! pip install g2papi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import json\n",
    "#os.listdir('../../../g2p/Genomics2Proteins_portal/server/data')\n",
    "\n",
    "filepath = '../../../g2p/Genomics2Proteins_portal/server/data/g2p_metadata_december_2024.json'\n",
    "with open(filepath, 'r') as file:\n",
    "    data = json.load(file)\n",
    "\n",
    "\n",
    "uniprot_to_gene = {}\n",
    "for gene, entries in data.items():\n",
    "    for entry in entries:\n",
    "        uniprot = entry['UniprotKB_Entry']\n",
    "        if not uniprot in uniprot_to_gene:\n",
    "            uniprot_to_gene[uniprot] = []\n",
    "    uniprot_to_gene[uniprot].append(gene)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/14004 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 14004/14004 [34:00<00:00,  6.86it/s] \n"
     ]
    }
   ],
   "source": [
    "# Get genes and find associated transcript for canonical isoform of each:\n",
    "\n",
    "# 1. Call the g2p API to get the gene\n",
    "import g2papi\n",
    "\n",
    "uniprot_list_path = \"../data/uniprots.txt\" # newline separated list of uniprots\n",
    "uniprot_list = open(uniprot_list_path, 'r').read().splitlines()\n",
    "\n",
    "uniprot_transcript_dict = {}\n",
    "\n",
    "found = 0\n",
    "notfound = 0\n",
    "from tqdm import tqdm\n",
    "for uniprot in tqdm(uniprot_list):\n",
    "    if uniprot in uniprot_to_gene:\n",
    "        if len(uniprot_to_gene[uniprot]) < 1:\n",
    "            notfound += 1\n",
    "            continue\n",
    "\n",
    "        found += 1\n",
    "        gene = uniprot_to_gene[uniprot][0]\n",
    "        uniprot_transcript_dict[uniprot] = g2papi.get_gene_transcript_protein_isoform_structure(gene, uniprot)\n",
    "    else:\n",
    "        notfound += 1\n",
    "\n",
    "# This list has 14,000 genes.... probably should have checked that first\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "# make a giant dataframe (each entry in the list is a df, so just concat them all\n",
    "transcript_df = pd.concat([pd.DataFrame(uniprot_transcript_dict[uniprot]) for uniprot in uniprot_transcript_dict.keys()])\n",
    "\n",
    "# save the transcript df to a file\n",
    "transcript_df.to_csv('../data/uniprot_transcript_df.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11149 2855\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>UniProt Isoform</th>\n",
       "      <th>Ensembl Gene Id</th>\n",
       "      <th>Ensembl Protein Id</th>\n",
       "      <th>Ensembl Transcript Id</th>\n",
       "      <th>RefSeq mRNA Id</th>\n",
       "      <th>PDB Ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q9H400</td>\n",
       "      <td>Q9H400-1(*)</td>\n",
       "      <td>ENSG00000203896</td>\n",
       "      <td>ENSP00000309521</td>\n",
       "      <td>ENST00000309546(*)</td>\n",
       "      <td>NM_017806(MANE)</td>\n",
       "      <td>not-available</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q9H400</td>\n",
       "      <td>not-assigned</td>\n",
       "      <td>ENSG00000203896</td>\n",
       "      <td>ENSP00000477561</td>\n",
       "      <td>ENST00000480139</td>\n",
       "      <td>NM_001305654</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q9H400</td>\n",
       "      <td>not-assigned</td>\n",
       "      <td>ENSG00000203896</td>\n",
       "      <td>ENSP00000477561</td>\n",
       "      <td>ENST00000480139</td>\n",
       "      <td>NM_001305655</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q9NR71</td>\n",
       "      <td>Q9NR71-1(*)</td>\n",
       "      <td>ENSG00000188611</td>\n",
       "      <td>ENSP00000378897</td>\n",
       "      <td>ENST00000395526</td>\n",
       "      <td>not-assigned</td>\n",
       "      <td>4WGK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q9NR71</td>\n",
       "      <td>Q9NR71-1(*)</td>\n",
       "      <td>ENSG00000188611</td>\n",
       "      <td>ENSP00000506746</td>\n",
       "      <td>ENST00000682911(*)</td>\n",
       "      <td>NM_019893(MANE)</td>\n",
       "      <td>4WGK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniProtKB UniProt Isoform  Ensembl Gene Id Ensembl Protein Id  \\\n",
       "0    Q9H400     Q9H400-1(*)  ENSG00000203896    ENSP00000309521   \n",
       "1    Q9H400    not-assigned  ENSG00000203896    ENSP00000477561   \n",
       "2    Q9H400    not-assigned  ENSG00000203896    ENSP00000477561   \n",
       "0    Q9NR71     Q9NR71-1(*)  ENSG00000188611    ENSP00000378897   \n",
       "1    Q9NR71     Q9NR71-1(*)  ENSG00000188611    ENSP00000506746   \n",
       "\n",
       "  Ensembl Transcript Id   RefSeq mRNA Id        PDB Ids  \n",
       "0    ENST00000309546(*)  NM_017806(MANE)  not-available  \n",
       "1       ENST00000480139     NM_001305654            NaN  \n",
       "2       ENST00000480139     NM_001305655            NaN  \n",
       "0       ENST00000395526     not-assigned           4WGK  \n",
       "1    ENST00000682911(*)  NM_019893(MANE)           4WGK  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(notfound, found)\n",
    "transcript_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/d8/p2r3zx7x1_920ng69wrnpjsc0000gq/T/ipykernel_36320/614285695.py:3: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.\n",
      "  canonical_df = transcript_df[transcript_df['UniProt Isoform'].str.contains('(\\*)')]# ENST00000373020.9\n",
      "/var/folders/d8/p2r3zx7x1_920ng69wrnpjsc0000gq/T/ipykernel_36320/614285695.py:6: FutureWarning: The default value of regex will change from True to False in a future version.\n",
      "  canonical_df[\"Ensembl Transcript Id\"] = canonical_df[\"Ensembl Transcript Id\"].str.replace('\\(\\*\\)', '')\n",
      "/var/folders/d8/p2r3zx7x1_920ng69wrnpjsc0000gq/T/ipykernel_36320/614285695.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  canonical_df[\"Ensembl Transcript Id\"] = canonical_df[\"Ensembl Transcript Id\"].str.replace('\\(\\*\\)', '')\n"
     ]
    }
   ],
   "source": [
    "# filter where UniProt Isoform column has (*) (means it's a canonical isoform)\n",
    "transcript_df = transcript_df[transcript_df['UniProt Isoform'].notna()]\n",
    "canonical_df = transcript_df[transcript_df['UniProt Isoform'].str.contains('(\\*)')]# ENST00000373020.9\n",
    "\n",
    "# remove (*) from Ensembl Transcript Id\n",
    "canonical_df[\"Ensembl Transcript Id\"] = canonical_df[\"Ensembl Transcript Id\"].str.replace('\\(\\*\\)', '')\n",
    "\n",
    "\n",
    "transcripts = canonical_df['Ensembl Transcript Id'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|█████████▉| 244939/244940 [3:39:28<00:00, 18.60it/s]     \n"
     ]
    }
   ],
   "source": [
    "\n",
    "# gtex data is stored in a giant txt table, we can only read one line at a time. first line is header\n",
    "gtex_filepath = '../data/GTEx_Analysis_v10_RSEMv1.3.3_transcripts_tpm.txt'\n",
    "\n",
    "\n",
    "def get_completed_transcripts():\n",
    "    if os.path.exists('../data/gtex_filtered_transcripts.csv'):\n",
    "        return list(pd.read_csv('../data/gtex_filtered_transcripts.csv')['transcript_id'].unique())\n",
    "    else:\n",
    "        return []\n",
    "\n",
    "completed_transcripts = get_completed_transcripts()\n",
    "\n",
    "# add all tmp transcripts to completed_transcripts\n",
    "if not os.path.exists('../data/tmp'):\n",
    "    os.makedirs('../data/tmp')\n",
    "tmp_files = os.listdir('../data/tmp')\n",
    "tmp_transcripts = [file.split('_')[-1].split('.')[0] for file in tmp_files]\n",
    "completed_transcripts.extend(tmp_transcripts)\n",
    "completed_transcripts = set(completed_transcripts)\n",
    "transcripts = set(transcripts)\n",
    "\n",
    "# if there is a file already, load it\n",
    "if os.path.exists('../data/gtex_filtered_transcripts.csv'):\n",
    "    gtex_df = pd.read_csv('../data/gtex_filtered_transcripts.csv')\n",
    "\n",
    "# get the header\n",
    "from tqdm import tqdm\n",
    "with open(gtex_filepath, 'r') as file:\n",
    "    header = file.readline()\n",
    "    header = header.split('\\t')\n",
    "\n",
    "    # now loop through each line\n",
    "    for line in tqdm(file, total=244940):\n",
    "        # split the line by tab\n",
    "        fields = line.split('\\t')\n",
    "        # get the transcript id\n",
    "        transcript_id = fields[0].split('.')[0]\n",
    "        # if the transcript id is in the transcripts list, print the line\n",
    "        if transcript_id in transcripts and transcript_id not in completed_transcripts:\n",
    "            # add the line to the dataframe with pdconcat\n",
    "            new_row = pd.DataFrame([fields])\n",
    "            # save to tmp file\n",
    "            tmp_filename = f'../data/tmp/gtex_filtered_transcripts_{transcript_id}.csv'\n",
    "            if not os.path.exists(\"../data/tmp\"):\n",
    "                os.makedirs(\"../data/tmp\")\n",
    "            new_row.to_csv(tmp_filename, index=False)\n",
    "            #new_df = pd.concat([new_df, pd.DataFrame([fields])])\n",
    "\n",
    "import glob\n",
    "# first read all of the tmp files and concat them\n",
    "tmp_files = glob.glob('../data/tmp/*.csv')\n",
    "tmp_df = pd.concat([pd.read_csv(file) for file in tmp_files])\n",
    "\n",
    "\n",
    "gtex_df = pd.read_csv('../data/gtex_filtered_transcripts.csv')\n",
    "gtex_df = pd.concat([gtex_df, tmp_df])\n",
    "\n",
    "# remove tmp files\n",
    "for file in tmp_files:\n",
    "    os.remove(file)\n",
    "\n",
    "# save the gtex df to a file\n",
    "gtex_df.to_csv('../data/gtex_filtered_transcripts.csv', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1568, 19790)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GTEX-1117F-0005-SM-HL9SH</th>\n",
       "      <th>GTEX-1117F-0011-R10b-SM-GI4VE</th>\n",
       "      <th>GTEX-1117F-0011-R11b-SM-GIN8R</th>\n",
       "      <th>GTEX-1117F-0011-R2b-SM-GI4VL</th>\n",
       "      <th>GTEX-1117F-0011-R3a-SM-GJ3PJ</th>\n",
       "      <th>GTEX-1117F-0011-R4b-SM-GI4VM</th>\n",
       "      <th>GTEX-1117F-0011-R5a-SM-GI4VW</th>\n",
       "      <th>GTEX-1117F-0011-R6a-SM-GI4VX</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZZPU-1326-SM-5GZWS</th>\n",
       "      <th>GTEX-ZZPU-1426-SM-5GZZ6</th>\n",
       "      <th>GTEX-ZZPU-1826-SM-5E43L</th>\n",
       "      <th>GTEX-ZZPU-2126-SM-5EGIU</th>\n",
       "      <th>GTEX-ZZPU-2226-SM-5EGIV</th>\n",
       "      <th>GTEX-ZZPU-2326-SM-GOQYU</th>\n",
       "      <th>GTEX-ZZPU-2426-SM-5E44I</th>\n",
       "      <th>GTEX-ZZPU-2526-SM-GOQZ3</th>\n",
       "      <th>GTEX-ZZPU-2626-SM-5E45Y</th>\n",
       "      <th>GTEX-ZZPU-2726-SM-5NQ8O\\n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000522656.5</td>\n",
       "      <td>ENSG00000003989.18</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.09</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.44</td>\n",
       "      <td>0.8</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2.16</td>\n",
       "      <td>0.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000640220.1</td>\n",
       "      <td>ENSG00000003989.18</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000356634.7</td>\n",
       "      <td>ENSG00000004487.18</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.26</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.58</td>\n",
       "      <td>3.19</td>\n",
       "      <td>2.44</td>\n",
       "      <td>2.79</td>\n",
       "      <td>3.6</td>\n",
       "      <td>...</td>\n",
       "      <td>6.92</td>\n",
       "      <td>5.98</td>\n",
       "      <td>11.26</td>\n",
       "      <td>22.7</td>\n",
       "      <td>9.66</td>\n",
       "      <td>10.14</td>\n",
       "      <td>9.98</td>\n",
       "      <td>8.76</td>\n",
       "      <td>10.34</td>\n",
       "      <td>6.91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000421138.6</td>\n",
       "      <td>ENSG00000004700.16</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.27</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.68</td>\n",
       "      <td>...</td>\n",
       "      <td>2.08</td>\n",
       "      <td>2.65</td>\n",
       "      <td>3.65</td>\n",
       "      <td>1.97</td>\n",
       "      <td>3.21</td>\n",
       "      <td>2.02</td>\n",
       "      <td>4.8</td>\n",
       "      <td>2.78</td>\n",
       "      <td>3.97</td>\n",
       "      <td>2.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000621914.4</td>\n",
       "      <td>ENSG00000006125.18</td>\n",
       "      <td>0.78</td>\n",
       "      <td>19.38</td>\n",
       "      <td>7.35</td>\n",
       "      <td>15.56</td>\n",
       "      <td>15.48</td>\n",
       "      <td>9.09</td>\n",
       "      <td>18.4</td>\n",
       "      <td>21.18</td>\n",
       "      <td>...</td>\n",
       "      <td>16.67</td>\n",
       "      <td>8.09</td>\n",
       "      <td>14.55</td>\n",
       "      <td>22.14</td>\n",
       "      <td>6.04</td>\n",
       "      <td>12.52</td>\n",
       "      <td>9.84</td>\n",
       "      <td>8.13</td>\n",
       "      <td>2.49</td>\n",
       "      <td>12.8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 19790 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       transcript_id             gene_id GTEX-1117F-0005-SM-HL9SH  \\\n",
       "0  ENST00000522656.5  ENSG00000003989.18                      0.0   \n",
       "0  ENST00000640220.1  ENSG00000003989.18                      0.0   \n",
       "0  ENST00000356634.7  ENSG00000004487.18                      0.0   \n",
       "0  ENST00000421138.6  ENSG00000004700.16                     0.01   \n",
       "0  ENST00000621914.4  ENSG00000006125.18                     0.78   \n",
       "\n",
       "  GTEX-1117F-0011-R10b-SM-GI4VE GTEX-1117F-0011-R11b-SM-GIN8R  \\\n",
       "0                           0.0                           0.0   \n",
       "0                           0.0                           0.0   \n",
       "0                          3.26                           0.0   \n",
       "0                          0.19                          0.46   \n",
       "0                         19.38                          7.35   \n",
       "\n",
       "  GTEX-1117F-0011-R2b-SM-GI4VL GTEX-1117F-0011-R3a-SM-GJ3PJ  \\\n",
       "0                          0.0                         0.02   \n",
       "0                          0.0                          0.0   \n",
       "0                         1.58                         3.19   \n",
       "0                         0.31                         0.27   \n",
       "0                        15.56                        15.48   \n",
       "\n",
       "  GTEX-1117F-0011-R4b-SM-GI4VM GTEX-1117F-0011-R5a-SM-GI4VW  \\\n",
       "0                          0.0                         0.02   \n",
       "0                          0.0                          0.0   \n",
       "0                         2.44                         2.79   \n",
       "0                         1.14                         0.93   \n",
       "0                         9.09                         18.4   \n",
       "\n",
       "  GTEX-1117F-0011-R6a-SM-GI4VX  ... GTEX-ZZPU-1326-SM-5GZWS  \\\n",
       "0                          0.0  ...                     0.3   \n",
       "0                          0.0  ...                     0.0   \n",
       "0                          3.6  ...                    6.92   \n",
       "0                         0.68  ...                    2.08   \n",
       "0                        21.18  ...                   16.67   \n",
       "\n",
       "  GTEX-ZZPU-1426-SM-5GZZ6 GTEX-ZZPU-1826-SM-5E43L GTEX-ZZPU-2126-SM-5EGIU  \\\n",
       "0                     0.0                    0.09                    0.35   \n",
       "0                     0.0                     0.0                     0.0   \n",
       "0                    5.98                   11.26                    22.7   \n",
       "0                    2.65                    3.65                    1.97   \n",
       "0                    8.09                   14.55                   22.14   \n",
       "\n",
       "  GTEX-ZZPU-2226-SM-5EGIV GTEX-ZZPU-2326-SM-GOQYU GTEX-ZZPU-2426-SM-5E44I  \\\n",
       "0                     0.0                    0.44                     0.8   \n",
       "0                     0.0                     0.0                     0.0   \n",
       "0                    9.66                   10.14                    9.98   \n",
       "0                    3.21                    2.02                     4.8   \n",
       "0                    6.04                   12.52                    9.84   \n",
       "\n",
       "  GTEX-ZZPU-2526-SM-GOQZ3 GTEX-ZZPU-2626-SM-5E45Y GTEX-ZZPU-2726-SM-5NQ8O\\n  \n",
       "0                     0.1                    2.16                      0.88  \n",
       "0                     0.0                     0.0                       0.0  \n",
       "0                    8.76                   10.34                      6.91  \n",
       "0                    2.78                    3.97                      2.23  \n",
       "0                    8.13                    2.49                      12.8  \n",
       "\n",
       "[5 rows x 19790 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# convert header to column names by splitting on tab\n",
    "gtex_df.columns = header\n",
    "\n",
    "# strip /n from the end of every row and the column names\n",
    "gtex_df = gtex_df.applymap(lambda x: x.rstrip('\\n') if isinstance(x, str) else x)\n",
    "\n",
    "gtex_df.head()\n",
    "# save df to csv\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove /n from column names\n",
    "gtex_df.columns = gtex_df.columns.str.rstrip('\\n')\n",
    "\n",
    "gtex_df.to_csv('../data/gtex_filtered_transcripts.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GTEX-1117F-0005-SM-HL9SH</th>\n",
       "      <th>GTEX-1117F-0011-R10b-SM-GI4VE</th>\n",
       "      <th>GTEX-1117F-0011-R11b-SM-GIN8R</th>\n",
       "      <th>GTEX-1117F-0011-R2b-SM-GI4VL</th>\n",
       "      <th>GTEX-1117F-0011-R3a-SM-GJ3PJ</th>\n",
       "      <th>GTEX-1117F-0011-R4b-SM-GI4VM</th>\n",
       "      <th>GTEX-1117F-0011-R5a-SM-GI4VW</th>\n",
       "      <th>GTEX-1117F-0011-R6a-SM-GI4VX</th>\n",
       "      <th>...</th>\n",
       "      <th>19780</th>\n",
       "      <th>19781</th>\n",
       "      <th>19782</th>\n",
       "      <th>19783</th>\n",
       "      <th>19784</th>\n",
       "      <th>19785</th>\n",
       "      <th>19786</th>\n",
       "      <th>19787</th>\n",
       "      <th>19788</th>\n",
       "      <th>19789</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000522656</td>\n",
       "      <td>ENSG00000003989.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000640220</td>\n",
       "      <td>ENSG00000003989.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000356634</td>\n",
       "      <td>ENSG00000004487.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.26</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.58</td>\n",
       "      <td>3.19</td>\n",
       "      <td>2.44</td>\n",
       "      <td>2.79</td>\n",
       "      <td>3.60</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000421138</td>\n",
       "      <td>ENSG00000004700.16</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.27</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.68</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000621914</td>\n",
       "      <td>ENSG00000006125.18</td>\n",
       "      <td>0.78</td>\n",
       "      <td>19.38</td>\n",
       "      <td>7.35</td>\n",
       "      <td>15.56</td>\n",
       "      <td>15.48</td>\n",
       "      <td>9.09</td>\n",
       "      <td>18.40</td>\n",
       "      <td>21.18</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 39580 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     transcript_id             gene_id  GTEX-1117F-0005-SM-HL9SH  \\\n",
       "0  ENST00000522656  ENSG00000003989.18                      0.00   \n",
       "1  ENST00000640220  ENSG00000003989.18                      0.00   \n",
       "2  ENST00000356634  ENSG00000004487.18                      0.00   \n",
       "3  ENST00000421138  ENSG00000004700.16                      0.01   \n",
       "4  ENST00000621914  ENSG00000006125.18                      0.78   \n",
       "\n",
       "   GTEX-1117F-0011-R10b-SM-GI4VE  GTEX-1117F-0011-R11b-SM-GIN8R  \\\n",
       "0                           0.00                           0.00   \n",
       "1                           0.00                           0.00   \n",
       "2                           3.26                           0.00   \n",
       "3                           0.19                           0.46   \n",
       "4                          19.38                           7.35   \n",
       "\n",
       "   GTEX-1117F-0011-R2b-SM-GI4VL  GTEX-1117F-0011-R3a-SM-GJ3PJ  \\\n",
       "0                          0.00                          0.02   \n",
       "1                          0.00                          0.00   \n",
       "2                          1.58                          3.19   \n",
       "3                          0.31                          0.27   \n",
       "4                         15.56                         15.48   \n",
       "\n",
       "   GTEX-1117F-0011-R4b-SM-GI4VM  GTEX-1117F-0011-R5a-SM-GI4VW  \\\n",
       "0                          0.00                          0.02   \n",
       "1                          0.00                          0.00   \n",
       "2                          2.44                          2.79   \n",
       "3                          1.14                          0.93   \n",
       "4                          9.09                         18.40   \n",
       "\n",
       "   GTEX-1117F-0011-R6a-SM-GI4VX  ...  19780  19781  19782  19783  19784  \\\n",
       "0                          0.00  ...    NaN    NaN    NaN    NaN    NaN   \n",
       "1                          0.00  ...    NaN    NaN    NaN    NaN    NaN   \n",
       "2                          3.60  ...    NaN    NaN    NaN    NaN    NaN   \n",
       "3                          0.68  ...    NaN    NaN    NaN    NaN    NaN   \n",
       "4                         21.18  ...    NaN    NaN    NaN    NaN    NaN   \n",
       "\n",
       "   19785  19786  19787  19788  19789  \n",
       "0    NaN    NaN    NaN    NaN    NaN  \n",
       "1    NaN    NaN    NaN    NaN    NaN  \n",
       "2    NaN    NaN    NaN    NaN    NaN  \n",
       "3    NaN    NaN    NaN    NaN    NaN  \n",
       "4    NaN    NaN    NaN    NaN    NaN  \n",
       "\n",
       "[5 rows x 39580 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# ok, so we have 2 things. we need to figure out what those ids mean in the column names, \n",
    "# and we need to add a gene name column\n",
    "\n",
    "gtex_df[\"transcript_id\"] = gtex_df[\"transcript_id\"].str.split('.').str[0]\n",
    "gtex_df.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>UniProt Isoform</th>\n",
       "      <th>Ensembl Gene Id</th>\n",
       "      <th>Ensembl Protein Id</th>\n",
       "      <th>Ensembl Transcript Id</th>\n",
       "      <th>RefSeq mRNA Id</th>\n",
       "      <th>PDB Ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q9H400</td>\n",
       "      <td>Q9H400-1(*)</td>\n",
       "      <td>ENSG00000203896</td>\n",
       "      <td>ENSP00000309521</td>\n",
       "      <td>ENST00000309546</td>\n",
       "      <td>NM_017806(MANE)</td>\n",
       "      <td>not-available</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q9NR71</td>\n",
       "      <td>Q9NR71-1(*)</td>\n",
       "      <td>ENSG00000188611</td>\n",
       "      <td>ENSP00000378897</td>\n",
       "      <td>ENST00000395526</td>\n",
       "      <td>not-assigned</td>\n",
       "      <td>4WGK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q9NR71</td>\n",
       "      <td>Q9NR71-1(*)</td>\n",
       "      <td>ENSG00000188611</td>\n",
       "      <td>ENSP00000506746</td>\n",
       "      <td>ENST00000682911</td>\n",
       "      <td>NM_019893(MANE)</td>\n",
       "      <td>4WGK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q86XT9</td>\n",
       "      <td>Q86XT9(*)</td>\n",
       "      <td>ENSG00000149932</td>\n",
       "      <td>ENSP00000279396</td>\n",
       "      <td>ENST00000279396</td>\n",
       "      <td>NM_001083613(MANE)</td>\n",
       "      <td>not-available</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q86XT9</td>\n",
       "      <td>Q86XT9(*)</td>\n",
       "      <td>ENSG00000149932</td>\n",
       "      <td>ENSP00000279396</td>\n",
       "      <td>ENST00000279396</td>\n",
       "      <td>NM_001369688</td>\n",
       "      <td>not-available</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniProtKB UniProt Isoform  Ensembl Gene Id Ensembl Protein Id  \\\n",
       "0    Q9H400     Q9H400-1(*)  ENSG00000203896    ENSP00000309521   \n",
       "0    Q9NR71     Q9NR71-1(*)  ENSG00000188611    ENSP00000378897   \n",
       "1    Q9NR71     Q9NR71-1(*)  ENSG00000188611    ENSP00000506746   \n",
       "0    Q86XT9       Q86XT9(*)  ENSG00000149932    ENSP00000279396   \n",
       "1    Q86XT9       Q86XT9(*)  ENSG00000149932    ENSP00000279396   \n",
       "\n",
       "  Ensembl Transcript Id      RefSeq mRNA Id        PDB Ids  \n",
       "0       ENST00000309546     NM_017806(MANE)  not-available  \n",
       "0       ENST00000395526        not-assigned           4WGK  \n",
       "1       ENST00000682911     NM_019893(MANE)           4WGK  \n",
       "0       ENST00000279396  NM_001083613(MANE)  not-available  \n",
       "1       ENST00000279396        NM_001369688  not-available  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "canonical_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "# join canonical df to gtex_df on \"Ensembl Transcript Id\" (canonical df) and transcript_id (gtex_df)\n",
    "gtex_gene_df = gtex_df.merge(canonical_df, left_on=\"transcript_id\", right_on=\"Ensembl Transcript Id\", how=\"left\")\n",
    "\n",
    "\n",
    "gtex_gene_df.to_csv('../data/gtex_gene_df.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>GTEX-1117F-0005-SM-HL9SH</th>\n",
       "      <th>GTEX-1117F-0011-R10b-SM-GI4VE</th>\n",
       "      <th>GTEX-1117F-0011-R11b-SM-GIN8R</th>\n",
       "      <th>GTEX-1117F-0011-R2b-SM-GI4VL</th>\n",
       "      <th>GTEX-1117F-0011-R3a-SM-GJ3PJ</th>\n",
       "      <th>GTEX-1117F-0011-R4b-SM-GI4VM</th>\n",
       "      <th>GTEX-1117F-0011-R5a-SM-GI4VW</th>\n",
       "      <th>GTEX-1117F-0011-R6a-SM-GI4VX</th>\n",
       "      <th>...</th>\n",
       "      <th>19787</th>\n",
       "      <th>19788</th>\n",
       "      <th>19789</th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>UniProt Isoform</th>\n",
       "      <th>Ensembl Gene Id</th>\n",
       "      <th>Ensembl Protein Id</th>\n",
       "      <th>Ensembl Transcript Id</th>\n",
       "      <th>RefSeq mRNA Id</th>\n",
       "      <th>PDB Ids</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENST00000522656</td>\n",
       "      <td>ENSG00000003989.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.02</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P52569</td>\n",
       "      <td>P52569-1(*)</td>\n",
       "      <td>ENSG00000003989</td>\n",
       "      <td>ENSP00000430464</td>\n",
       "      <td>ENST00000522656</td>\n",
       "      <td>not-assigned</td>\n",
       "      <td>not-available</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENST00000640220</td>\n",
       "      <td>ENSG00000003989.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P52569</td>\n",
       "      <td>P52569-1(*)</td>\n",
       "      <td>ENSG00000003989</td>\n",
       "      <td>ENSP00000492016</td>\n",
       "      <td>ENST00000640220</td>\n",
       "      <td>not-assigned</td>\n",
       "      <td>not-available</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENST00000356634</td>\n",
       "      <td>ENSG00000004487.18</td>\n",
       "      <td>0.00</td>\n",
       "      <td>3.26</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.58</td>\n",
       "      <td>3.19</td>\n",
       "      <td>2.44</td>\n",
       "      <td>2.79</td>\n",
       "      <td>3.60</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O60341</td>\n",
       "      <td>O60341-1(*)</td>\n",
       "      <td>ENSG00000004487</td>\n",
       "      <td>ENSP00000349049</td>\n",
       "      <td>ENST00000356634</td>\n",
       "      <td>NM_015013</td>\n",
       "      <td>2COM,2DW4,2EJR,2H94,2HKO,2IW5,2L3D,2UXN,2UXX,2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENST00000421138</td>\n",
       "      <td>ENSG00000004700.16</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0.19</td>\n",
       "      <td>0.46</td>\n",
       "      <td>0.31</td>\n",
       "      <td>0.27</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.93</td>\n",
       "      <td>0.68</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P46063</td>\n",
       "      <td>P46063(*)</td>\n",
       "      <td>ENSG00000004700</td>\n",
       "      <td>ENSP00000395449</td>\n",
       "      <td>ENST00000421138</td>\n",
       "      <td>not-assigned</td>\n",
       "      <td>2V1X,2WWY,4U7D,6JTZ,8YRS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENST00000621914</td>\n",
       "      <td>ENSG00000006125.18</td>\n",
       "      <td>0.78</td>\n",
       "      <td>19.38</td>\n",
       "      <td>7.35</td>\n",
       "      <td>15.56</td>\n",
       "      <td>15.48</td>\n",
       "      <td>9.09</td>\n",
       "      <td>18.40</td>\n",
       "      <td>21.18</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P63010</td>\n",
       "      <td>P63010-1(*)</td>\n",
       "      <td>ENSG00000006125</td>\n",
       "      <td>ENSP00000482315</td>\n",
       "      <td>ENST00000621914</td>\n",
       "      <td>NM_001282</td>\n",
       "      <td>1E42,2G30,2IV8,2IV9,2JKR,2JKT,2VGL,2XA7,4UQI,5...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 39587 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     transcript_id             gene_id  GTEX-1117F-0005-SM-HL9SH  \\\n",
       "0  ENST00000522656  ENSG00000003989.18                      0.00   \n",
       "1  ENST00000640220  ENSG00000003989.18                      0.00   \n",
       "2  ENST00000356634  ENSG00000004487.18                      0.00   \n",
       "3  ENST00000421138  ENSG00000004700.16                      0.01   \n",
       "4  ENST00000621914  ENSG00000006125.18                      0.78   \n",
       "\n",
       "   GTEX-1117F-0011-R10b-SM-GI4VE  GTEX-1117F-0011-R11b-SM-GIN8R  \\\n",
       "0                           0.00                           0.00   \n",
       "1                           0.00                           0.00   \n",
       "2                           3.26                           0.00   \n",
       "3                           0.19                           0.46   \n",
       "4                          19.38                           7.35   \n",
       "\n",
       "   GTEX-1117F-0011-R2b-SM-GI4VL  GTEX-1117F-0011-R3a-SM-GJ3PJ  \\\n",
       "0                          0.00                          0.02   \n",
       "1                          0.00                          0.00   \n",
       "2                          1.58                          3.19   \n",
       "3                          0.31                          0.27   \n",
       "4                         15.56                         15.48   \n",
       "\n",
       "   GTEX-1117F-0011-R4b-SM-GI4VM  GTEX-1117F-0011-R5a-SM-GI4VW  \\\n",
       "0                          0.00                          0.02   \n",
       "1                          0.00                          0.00   \n",
       "2                          2.44                          2.79   \n",
       "3                          1.14                          0.93   \n",
       "4                          9.09                         18.40   \n",
       "\n",
       "   GTEX-1117F-0011-R6a-SM-GI4VX  ...  19787  19788  19789  UniProtKB  \\\n",
       "0                          0.00  ...    NaN    NaN    NaN     P52569   \n",
       "1                          0.00  ...    NaN    NaN    NaN     P52569   \n",
       "2                          3.60  ...    NaN    NaN    NaN     O60341   \n",
       "3                          0.68  ...    NaN    NaN    NaN     P46063   \n",
       "4                         21.18  ...    NaN    NaN    NaN     P63010   \n",
       "\n",
       "   UniProt Isoform  Ensembl Gene Id  Ensembl Protein Id  \\\n",
       "0      P52569-1(*)  ENSG00000003989     ENSP00000430464   \n",
       "1      P52569-1(*)  ENSG00000003989     ENSP00000492016   \n",
       "2      O60341-1(*)  ENSG00000004487     ENSP00000349049   \n",
       "3        P46063(*)  ENSG00000004700     ENSP00000395449   \n",
       "4      P63010-1(*)  ENSG00000006125     ENSP00000482315   \n",
       "\n",
       "   Ensembl Transcript Id  RefSeq mRNA Id  \\\n",
       "0        ENST00000522656    not-assigned   \n",
       "1        ENST00000640220    not-assigned   \n",
       "2        ENST00000356634       NM_015013   \n",
       "3        ENST00000421138    not-assigned   \n",
       "4        ENST00000621914       NM_001282   \n",
       "\n",
       "                                             PDB Ids  \n",
       "0                                      not-available  \n",
       "1                                      not-available  \n",
       "2  2COM,2DW4,2EJR,2H94,2HKO,2IW5,2L3D,2UXN,2UXX,2...  \n",
       "3                           2V1X,2WWY,4U7D,6JTZ,8YRS  \n",
       "4  1E42,2G30,2IV8,2IV9,2JKR,2JKT,2VGL,2XA7,4UQI,5...  \n",
       "\n",
       "[5 rows x 39587 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_gene_df.head()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load the GTEX-id attribute map to a df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARNAME</th>\n",
       "      <th>VARDESC</th>\n",
       "      <th>ACCESS</th>\n",
       "      <th>DOCFILE</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>UNITS</th>\n",
       "      <th>COMMENT1</th>\n",
       "      <th>COMMENT2</th>\n",
       "      <th>VALUES</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 48</th>\n",
       "      <th>Unnamed: 49</th>\n",
       "      <th>Unnamed: 50</th>\n",
       "      <th>Unnamed: 51</th>\n",
       "      <th>Unnamed: 52</th>\n",
       "      <th>Unnamed: 53</th>\n",
       "      <th>Unnamed: 54</th>\n",
       "      <th>Unnamed: 55</th>\n",
       "      <th>Unnamed: 56</th>\n",
       "      <th>Unnamed: 57</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMPID</td>\n",
       "      <td>Sample ID, GTEx Public Sample ID</td>\n",
       "      <td>Open</td>\n",
       "      <td>NaN</td>\n",
       "      <td>string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SMATSSCR</td>\n",
       "      <td>Autolysis Score</td>\n",
       "      <td>Open</td>\n",
       "      <td>PRC Case Summary Report</td>\n",
       "      <td>integer, encoded value</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Autolysis</td>\n",
       "      <td>The destruction of organism cells or tissues b...</td>\n",
       "      <td>0=None</td>\n",
       "      <td>1=Mild</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SMNABTCH</td>\n",
       "      <td>Nucleic Acid Isolation Batch ID</td>\n",
       "      <td>Open</td>\n",
       "      <td>LDACC</td>\n",
       "      <td>string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Generated at LDACC</td>\n",
       "      <td>Batch when DNA/RNA was isolated and extracted ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMNABTCHT</td>\n",
       "      <td>Type of nucleic acid isolation batch</td>\n",
       "      <td>Open</td>\n",
       "      <td>LDACC</td>\n",
       "      <td>string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Generated at LDACC</td>\n",
       "      <td>The process by which DNA/RNA was isolated</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SMNABTCHD</td>\n",
       "      <td>Date of nucleic acid isolation batch</td>\n",
       "      <td>Open</td>\n",
       "      <td>LDACC</td>\n",
       "      <td>string</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Generated at LDACC</td>\n",
       "      <td>The date on which DNA/RNA was isolated</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     VARNAME                               VARDESC ACCESS  \\\n",
       "0     SAMPID      Sample ID, GTEx Public Sample ID   Open   \n",
       "1   SMATSSCR                       Autolysis Score   Open   \n",
       "2   SMNABTCH       Nucleic Acid Isolation Batch ID   Open   \n",
       "3  SMNABTCHT  Type of nucleic acid isolation batch   Open   \n",
       "4  SMNABTCHD  Date of nucleic acid isolation batch   Open   \n",
       "\n",
       "                   DOCFILE                    TYPE UNITS            COMMENT1  \\\n",
       "0                      NaN                  string   NaN                 NaN   \n",
       "1  PRC Case Summary Report  integer, encoded value   NaN           Autolysis   \n",
       "2                    LDACC                  string   NaN  Generated at LDACC   \n",
       "3                    LDACC                  string   NaN  Generated at LDACC   \n",
       "4                    LDACC                  string   NaN  Generated at LDACC   \n",
       "\n",
       "                                            COMMENT2  VALUES Unnamed: 9  ...  \\\n",
       "0                                                NaN     NaN        NaN  ...   \n",
       "1  The destruction of organism cells or tissues b...  0=None     1=Mild  ...   \n",
       "2  Batch when DNA/RNA was isolated and extracted ...     NaN        NaN  ...   \n",
       "3          The process by which DNA/RNA was isolated     NaN        NaN  ...   \n",
       "4             The date on which DNA/RNA was isolated     NaN        NaN  ...   \n",
       "\n",
       "  Unnamed: 48 Unnamed: 49 Unnamed: 50 Unnamed: 51 Unnamed: 52 Unnamed: 53  \\\n",
       "0         NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "1         NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "2         NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "3         NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "4         NaN         NaN         NaN         NaN         NaN         NaN   \n",
       "\n",
       "  Unnamed: 54 Unnamed: 55 Unnamed: 56 Unnamed: 57  \n",
       "0         NaN         NaN         NaN         NaN  \n",
       "1         NaN         NaN         NaN         NaN  \n",
       "2         NaN         NaN         NaN         NaN  \n",
       "3         NaN         NaN         NaN         NaN  \n",
       "4         NaN         NaN         NaN         NaN  \n",
       "\n",
       "[5 rows x 58 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# GTEx_Analysis_v10_Annotations_SampleAttributesDD.xlsx\n",
    "# GTEx_Analysis_v10_Annotations_SampleAttributesDS.txt\n",
    "\n",
    "# load the xlsx file\n",
    "import pandas as pd\n",
    "\n",
    "gtex_col_names = pd.read_excel('../data/GTEx_Analysis_v10_Annotations_SampleAttributesDD.xlsx')\n",
    "gtex_col_names.head()\n",
    "\n",
    "# these are description of each column found in the txt file\n",
    "# the column names are in the VARNAME column\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/d8/p2r3zx7x1_920ng69wrnpjsc0000gq/T/ipykernel_3257/67884798.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  gtex_id_map = pd.read_csv('../data/GTEx_Analysis_v10_Annotations_SampleAttributesDS.txt', sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# load the txt file\n",
    "gtex_id_map = pd.read_csv('../data/GTEx_Analysis_v10_Annotations_SampleAttributesDS.txt', sep='\\t')\n",
    "# now set the column names to the VARNAME column\n",
    "\n",
    "# first filter to only include rows where ACCESS is Open\n",
    "#colnames = gtex_col_names[gtex_col_names['ACCESS'] == 'Open']\n",
    "\n",
    "# now set the column names to the VARNAME column\n",
    "#gtex_id_map.columns = colnames['VARNAME']\n",
    "gtex_id_map.head()\n",
    "\n",
    "\n",
    "# save first 10 rows to csv\n",
    "gtex_id_map.head(10).to_csv('../data/gtex_id_map_sample.csv', index=False)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/d8/p2r3zx7x1_920ng69wrnpjsc0000gq/T/ipykernel_3257/2728978780.py:2: DtypeWarning: Columns (0,1,19790,19791,39580,39581,39582,39583,39584,39585,39586) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  gtex_gene_df = pd.read_csv('../data/gtex_gene_df.csv')\n",
      "100%|██████████| 31/31 [00:04<00:00,  6.62it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Blood Vessel</th>\n",
       "      <th>Bladder</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Adrenal Gland</th>\n",
       "      <th>Adipose Tissue</th>\n",
       "      <th>Nerve</th>\n",
       "      <th>Colon</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>...</th>\n",
       "      <th>Breast</th>\n",
       "      <th>Blood</th>\n",
       "      <th>Small Intestine</th>\n",
       "      <th>Esophagus</th>\n",
       "      <th>Salivary Gland</th>\n",
       "      <th>Ovary</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Skin</th>\n",
       "      <th>Cervix Uteri</th>\n",
       "      <th>Prostate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.293159</td>\n",
       "      <td>0.101818</td>\n",
       "      <td>0.173298</td>\n",
       "      <td>0.114706</td>\n",
       "      <td>0.067695</td>\n",
       "      <td>0.292483</td>\n",
       "      <td>0.316149</td>\n",
       "      <td>0.048486</td>\n",
       "      <td>0.854817</td>\n",
       "      <td>0.163918</td>\n",
       "      <td>...</td>\n",
       "      <td>0.159786</td>\n",
       "      <td>0.000584</td>\n",
       "      <td>0.031416</td>\n",
       "      <td>0.064854</td>\n",
       "      <td>0.048453</td>\n",
       "      <td>0.691347</td>\n",
       "      <td>0.007906</td>\n",
       "      <td>0.212674</td>\n",
       "      <td>0.160851</td>\n",
       "      <td>0.052872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001186</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000442</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7.575737</td>\n",
       "      <td>11.469610</td>\n",
       "      <td>1.708262</td>\n",
       "      <td>9.404471</td>\n",
       "      <td>7.957661</td>\n",
       "      <td>7.122098</td>\n",
       "      <td>7.718284</td>\n",
       "      <td>9.081632</td>\n",
       "      <td>8.801467</td>\n",
       "      <td>7.791316</td>\n",
       "      <td>...</td>\n",
       "      <td>7.312082</td>\n",
       "      <td>6.774513</td>\n",
       "      <td>7.335354</td>\n",
       "      <td>8.761103</td>\n",
       "      <td>9.931547</td>\n",
       "      <td>12.081036</td>\n",
       "      <td>5.650903</td>\n",
       "      <td>11.599183</td>\n",
       "      <td>12.103191</td>\n",
       "      <td>7.573723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3.496031</td>\n",
       "      <td>3.208312</td>\n",
       "      <td>1.380213</td>\n",
       "      <td>1.992824</td>\n",
       "      <td>2.706169</td>\n",
       "      <td>3.062014</td>\n",
       "      <td>1.889000</td>\n",
       "      <td>2.029686</td>\n",
       "      <td>2.452579</td>\n",
       "      <td>2.456637</td>\n",
       "      <td>...</td>\n",
       "      <td>2.727393</td>\n",
       "      <td>4.873956</td>\n",
       "      <td>2.655531</td>\n",
       "      <td>2.181781</td>\n",
       "      <td>3.188066</td>\n",
       "      <td>2.072383</td>\n",
       "      <td>1.228736</td>\n",
       "      <td>5.991142</td>\n",
       "      <td>2.325106</td>\n",
       "      <td>1.301738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15.247477</td>\n",
       "      <td>21.453896</td>\n",
       "      <td>3.794929</td>\n",
       "      <td>16.870353</td>\n",
       "      <td>12.504068</td>\n",
       "      <td>16.495081</td>\n",
       "      <td>21.040881</td>\n",
       "      <td>16.834919</td>\n",
       "      <td>2.301834</td>\n",
       "      <td>22.358246</td>\n",
       "      <td>...</td>\n",
       "      <td>17.214125</td>\n",
       "      <td>9.471283</td>\n",
       "      <td>14.669912</td>\n",
       "      <td>16.441407</td>\n",
       "      <td>14.494641</td>\n",
       "      <td>26.117979</td>\n",
       "      <td>14.624729</td>\n",
       "      <td>18.321055</td>\n",
       "      <td>20.259149</td>\n",
       "      <td>16.677553</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Blood Vessel    Bladder     Liver     Vagina  Adrenal Gland  \\\n",
       "0      0.293159   0.101818  0.173298   0.114706       0.067695   \n",
       "1      0.000000   0.000000  0.000000   0.000000       0.000000   \n",
       "2      7.575737  11.469610  1.708262   9.404471       7.957661   \n",
       "3      3.496031   3.208312  1.380213   1.992824       2.706169   \n",
       "4     15.247477  21.453896  3.794929  16.870353      12.504068   \n",
       "\n",
       "   Adipose Tissue      Nerve      Colon    Muscle    Thyroid  ...     Breast  \\\n",
       "0        0.292483   0.316149   0.048486  0.854817   0.163918  ...   0.159786   \n",
       "1        0.000000   0.000000   0.000000  0.000000   0.000000  ...   0.000000   \n",
       "2        7.122098   7.718284   9.081632  8.801467   7.791316  ...   7.312082   \n",
       "3        3.062014   1.889000   2.029686  2.452579   2.456637  ...   2.727393   \n",
       "4       16.495081  21.040881  16.834919  2.301834  22.358246  ...  17.214125   \n",
       "\n",
       "      Blood  Small Intestine  Esophagus  Salivary Gland      Ovary     Spleen  \\\n",
       "0  0.000584         0.031416   0.064854        0.048453   0.691347   0.007906   \n",
       "1  0.001186         0.000000   0.000000        0.000000   0.000000   0.000000   \n",
       "2  6.774513         7.335354   8.761103        9.931547  12.081036   5.650903   \n",
       "3  4.873956         2.655531   2.181781        3.188066   2.072383   1.228736   \n",
       "4  9.471283        14.669912  16.441407       14.494641  26.117979  14.624729   \n",
       "\n",
       "        Skin  Cervix Uteri   Prostate  \n",
       "0   0.212674      0.160851   0.052872  \n",
       "1   0.000442      0.000000   0.000000  \n",
       "2  11.599183     12.103191   7.573723  \n",
       "3   5.991142      2.325106   1.301738  \n",
       "4  18.321055     20.259149  16.677553  \n",
       "\n",
       "[5 rows x 31 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load the gtex_gene_df\n",
    "gtex_gene_df = pd.read_csv('../data/gtex_gene_df.csv')\n",
    "\n",
    "smts_map = gtex_id_map['SMTS']\n",
    "\n",
    "# get the unique tissue types by making a set of the smts_map\n",
    "tissue_types = set(smts_map)\n",
    "\n",
    "# make a new gene_df with a column for each tissue type\n",
    "# get all cols for each tissue type from gtex_gene_df, and then take the mean of each\n",
    "tissue_wise_df = pd.DataFrame()\n",
    "from tqdm import tqdm\n",
    "for tissue in tqdm(tissue_types):\n",
    "    # first get all gtex_id_map rows where SMTS is the tissue type\n",
    "    tissue_rows = gtex_id_map[gtex_id_map['SMTS'] == tissue]\n",
    "\n",
    "    # get the corresponding columns from gtex_gene_df\n",
    "    tissue_sample_ids = tissue_rows['SAMPID']\n",
    "    tissue_cols = gtex_gene_df.columns[gtex_gene_df.columns.isin(tissue_sample_ids)]\n",
    "    tissue_df = gtex_gene_df[tissue_cols]\n",
    "\n",
    "    # now take the mean of each column\n",
    "    tissue_df = tissue_df.mean(axis=1)\n",
    "    tissue_wise_df[tissue] = tissue_df\n",
    "\n",
    "# save tissue_wise_df to csv\n",
    "tissue_wise_df.to_csv('../data/tissue_wise_df.csv', index=False)\n",
    "\n",
    "tissue_wise_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6217, 31)\n",
      "(6217, 39587)\n",
      "Index(['transcript_id', 'gene_id', 'GTEX-1117F-0005-SM-HL9SH',\n",
      "       'GTEX-1117F-0011-R10b-SM-GI4VE', 'GTEX-1117F-0011-R11b-SM-GIN8R',\n",
      "       'GTEX-1117F-0011-R2b-SM-GI4VL', 'GTEX-1117F-0011-R3a-SM-GJ3PJ',\n",
      "       'GTEX-1117F-0011-R4b-SM-GI4VM', 'GTEX-1117F-0011-R5a-SM-GI4VW',\n",
      "       'GTEX-1117F-0011-R6a-SM-GI4VX',\n",
      "       ...\n",
      "       '19787', '19788', '19789', 'UniProtKB', 'UniProt Isoform',\n",
      "       'Ensembl Gene Id', 'Ensembl Protein Id', 'Ensembl Transcript Id',\n",
      "       'RefSeq mRNA Id', 'PDB Ids'],\n",
      "      dtype='object', length=39587)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Blood Vessel</th>\n",
       "      <th>Bladder</th>\n",
       "      <th>Liver</th>\n",
       "      <th>Vagina</th>\n",
       "      <th>Adrenal Gland</th>\n",
       "      <th>Adipose Tissue</th>\n",
       "      <th>Nerve</th>\n",
       "      <th>Colon</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>...</th>\n",
       "      <th>Spleen</th>\n",
       "      <th>Skin</th>\n",
       "      <th>Cervix Uteri</th>\n",
       "      <th>Prostate</th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>UniProt Isoform</th>\n",
       "      <th>Ensembl Gene Id</th>\n",
       "      <th>Ensembl Protein Id</th>\n",
       "      <th>Ensembl Transcript Id</th>\n",
       "      <th>RefSeq mRNA Id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.293159</td>\n",
       "      <td>0.101818</td>\n",
       "      <td>0.173298</td>\n",
       "      <td>0.114706</td>\n",
       "      <td>0.067695</td>\n",
       "      <td>0.292483</td>\n",
       "      <td>0.316149</td>\n",
       "      <td>0.048486</td>\n",
       "      <td>0.854817</td>\n",
       "      <td>0.163918</td>\n",
       "      <td>...</td>\n",
       "      <td>0.007906</td>\n",
       "      <td>0.212674</td>\n",
       "      <td>0.160851</td>\n",
       "      <td>0.052872</td>\n",
       "      <td>P52569</td>\n",
       "      <td>P52569-1(*)</td>\n",
       "      <td>ENSG00000003989</td>\n",
       "      <td>ENSP00000430464</td>\n",
       "      <td>ENST00000522656</td>\n",
       "      <td>not-assigned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000442</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>P52569</td>\n",
       "      <td>P52569-1(*)</td>\n",
       "      <td>ENSG00000003989</td>\n",
       "      <td>ENSP00000492016</td>\n",
       "      <td>ENST00000640220</td>\n",
       "      <td>not-assigned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7.575737</td>\n",
       "      <td>11.469610</td>\n",
       "      <td>1.708262</td>\n",
       "      <td>9.404471</td>\n",
       "      <td>7.957661</td>\n",
       "      <td>7.122098</td>\n",
       "      <td>7.718284</td>\n",
       "      <td>9.081632</td>\n",
       "      <td>8.801467</td>\n",
       "      <td>7.791316</td>\n",
       "      <td>...</td>\n",
       "      <td>5.650903</td>\n",
       "      <td>11.599183</td>\n",
       "      <td>12.103191</td>\n",
       "      <td>7.573723</td>\n",
       "      <td>O60341</td>\n",
       "      <td>O60341-1(*)</td>\n",
       "      <td>ENSG00000004487</td>\n",
       "      <td>ENSP00000349049</td>\n",
       "      <td>ENST00000356634</td>\n",
       "      <td>NM_015013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3.496031</td>\n",
       "      <td>3.208312</td>\n",
       "      <td>1.380213</td>\n",
       "      <td>1.992824</td>\n",
       "      <td>2.706169</td>\n",
       "      <td>3.062014</td>\n",
       "      <td>1.889000</td>\n",
       "      <td>2.029686</td>\n",
       "      <td>2.452579</td>\n",
       "      <td>2.456637</td>\n",
       "      <td>...</td>\n",
       "      <td>1.228736</td>\n",
       "      <td>5.991142</td>\n",
       "      <td>2.325106</td>\n",
       "      <td>1.301738</td>\n",
       "      <td>P46063</td>\n",
       "      <td>P46063(*)</td>\n",
       "      <td>ENSG00000004700</td>\n",
       "      <td>ENSP00000395449</td>\n",
       "      <td>ENST00000421138</td>\n",
       "      <td>not-assigned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15.247477</td>\n",
       "      <td>21.453896</td>\n",
       "      <td>3.794929</td>\n",
       "      <td>16.870353</td>\n",
       "      <td>12.504068</td>\n",
       "      <td>16.495081</td>\n",
       "      <td>21.040881</td>\n",
       "      <td>16.834919</td>\n",
       "      <td>2.301834</td>\n",
       "      <td>22.358246</td>\n",
       "      <td>...</td>\n",
       "      <td>14.624729</td>\n",
       "      <td>18.321055</td>\n",
       "      <td>20.259149</td>\n",
       "      <td>16.677553</td>\n",
       "      <td>P63010</td>\n",
       "      <td>P63010-1(*)</td>\n",
       "      <td>ENSG00000006125</td>\n",
       "      <td>ENSP00000482315</td>\n",
       "      <td>ENST00000621914</td>\n",
       "      <td>NM_001282</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Blood Vessel    Bladder     Liver     Vagina  Adrenal Gland  \\\n",
       "0      0.293159   0.101818  0.173298   0.114706       0.067695   \n",
       "1      0.000000   0.000000  0.000000   0.000000       0.000000   \n",
       "2      7.575737  11.469610  1.708262   9.404471       7.957661   \n",
       "3      3.496031   3.208312  1.380213   1.992824       2.706169   \n",
       "4     15.247477  21.453896  3.794929  16.870353      12.504068   \n",
       "\n",
       "   Adipose Tissue      Nerve      Colon    Muscle    Thyroid  ...     Spleen  \\\n",
       "0        0.292483   0.316149   0.048486  0.854817   0.163918  ...   0.007906   \n",
       "1        0.000000   0.000000   0.000000  0.000000   0.000000  ...   0.000000   \n",
       "2        7.122098   7.718284   9.081632  8.801467   7.791316  ...   5.650903   \n",
       "3        3.062014   1.889000   2.029686  2.452579   2.456637  ...   1.228736   \n",
       "4       16.495081  21.040881  16.834919  2.301834  22.358246  ...  14.624729   \n",
       "\n",
       "        Skin  Cervix Uteri   Prostate  UniProtKB  UniProt Isoform  \\\n",
       "0   0.212674      0.160851   0.052872     P52569      P52569-1(*)   \n",
       "1   0.000442      0.000000   0.000000     P52569      P52569-1(*)   \n",
       "2  11.599183     12.103191   7.573723     O60341      O60341-1(*)   \n",
       "3   5.991142      2.325106   1.301738     P46063        P46063(*)   \n",
       "4  18.321055     20.259149  16.677553     P63010      P63010-1(*)   \n",
       "\n",
       "   Ensembl Gene Id  Ensembl Protein Id  Ensembl Transcript Id  RefSeq mRNA Id  \n",
       "0  ENSG00000003989     ENSP00000430464        ENST00000522656    not-assigned  \n",
       "1  ENSG00000003989     ENSP00000492016        ENST00000640220    not-assigned  \n",
       "2  ENSG00000004487     ENSP00000349049        ENST00000356634       NM_015013  \n",
       "3  ENSG00000004700     ENSP00000395449        ENST00000421138    not-assigned  \n",
       "4  ENSG00000006125     ENSP00000482315        ENST00000621914       NM_001282  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(tissue_wise_df.shape)\n",
    "print(gtex_gene_df.shape)\n",
    "print(gtex_gene_df.columns)\n",
    "\n",
    "# add the following columns to tissue_wise_df:\n",
    "# 'UniProtKB', 'UniProt Isoform',\n",
    "# 'Ensembl Gene Id', 'Ensembl Protein Id', 'Ensembl Transcript Id',\n",
    "# 'RefSeq mRNA Id'\n",
    "\n",
    "# get the columns from gtex_gene_df\n",
    "metadata_cols = ['UniProtKB', 'UniProt Isoform', 'Ensembl Gene Id', 'Ensembl Protein Id', 'Ensembl Transcript Id', 'RefSeq mRNA Id']\n",
    "\n",
    "# add the columns to tissue_wise_df\n",
    "tissue_wise_df = pd.concat([tissue_wise_df, gtex_gene_df[metadata_cols]], axis=1)\n",
    "tissue_wise_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6217, 37)\n"
     ]
    }
   ],
   "source": [
    "# print shape\n",
    "print(tissue_wise_df.shape)\n",
    "\n",
    "# save to csv\n",
    "tissue_wise_df.to_csv('../data/tissue_wise_df.csv', index=False)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "jupyter_default",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
